common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -251,15 +251,21 @@ describe 'Problem', ->
               callable()
             done: (callable) ->
               callable()
-        spyOn @problem, 'enableAllButtons'
         @problem.submit()
-        expect(@problem.enableAllButtons).toHaveBeenCalledWith false, true
         if jQuery.active == 0
           deferred.resolve()
         deferred.promise()
 
+      spyOn(@problem, 'disableButtons').and.callThrough()
```

Comment:
I'm not a fan of all this spying. Honestly, I don't care which methods were called as long as the state on the individual buttons is correct (which also makes the test more robust to code changes like this one).

I also want to make sure there is a test that fails without the bug fix. I'm not sure that is the case here, since it is just checking which methods are called.

Instead of spying, can you check the disabled attribute on the impacted buttons? This means that you may need to change the test case so that it will have a Save (and possibly Reset) button.